swagger_server/test/e2e/test_samples_get_ids_controller.py

```diff
@@ -0,0 +1,61 @@
+from hypothesis import given, strategies as st, reproduce_failure
+
+from swagger_server.test.strategies import samples, experiment_ids
+
+
+def test_samples_get_without_any_id(get_sample_by_ids):
+    assert get_sample_by_ids('').status_code == 404
+
+
+@given(sample_ids=st.lists(min_size=1, max_size=5, elements=experiment_ids()))
+def test_samples_get_non_existent_ids(sample_ids, get_sample_by_ids):
+    assert get_sample_by_ids(",".join(sample_ids)).status_code == 404
+
+
+@given(samples_with_ids=st.lists(min_size=1, max_size=5, elements=samples(), unique_by=lambda x: x.experiment_id))
+def test_samples_get_existing_ids(samples_with_ids, create_sample, create_leaf, get_sample_by_ids, sample_graph):
+    try:
+        created_ids = []
+        for sample in samples_with_ids:
+            if sample.nearest_neighbours:
+                for neighbour in sample.nearest_neighbours:
+                    create_sample(neighbour)
```

Comment:
You should put `ensure=True` in all of the fixture calls whose purpose is to setup the test and not the call under test itself, since we expect these to be successful.

Reply:
This test case could generate examples that multiple samples are inter-connected, so some samples may have already created when creating previous samples, which will cause conflict when trying to creating the same sample again.

But one thing I could do is to not generate any examples with samples that have neighbours, like next test case. The relationships among samples are not the purposes for this test case anyway. This could simplify the test case.

Reply:
> which will cause conflict when trying to creating the same sample again

Ah that's right. In this case it's totally valid.

Reply:
I have refactored the test cases to not generate neighbours and leaf nodes for the input samples. This should simply the test case: 74fe794